lib/components/SelectionControl/Checkbox/index.tsx

```diff
@@ -0,0 +1,170 @@
+import {
+    ComponentProps,
+    forwardRef,
+    ReactNode,
+    useState,
+    useEffect,
+} from "react";
+import clsx from "clsx";
+import "./checkbox.scss";
+import {
+    StandaloneSquareCheckFillIcon,
+    StandaloneSquareBoldIcon,
+    StandaloneSquareMinusFillIcon,
+    LabelPairedCircleInfoSmRegularIcon,
+    LabelPairedCircleInfoMdRegularIcon,
+} from "@deriv/quill-icons";
+import { TMediumSizes } from "../../../types";
+import { Text } from "../../Typography";
+
+interface CheckboxProps
+    extends Omit<
+        ComponentProps<"input">,
+        "placeholder" | "defaultChecked" | "size"
+    > {
+    indeterminate?: boolean;
+    showInfoIcon?: boolean;
+    infoIconClassName?: string;
+    size?: TMediumSizes;
+    label: ReactNode | string;
+    labelClassName?: string;
+    onChange?: (
+        e:
+            | React.ChangeEvent<HTMLInputElement>
+            | React.KeyboardEvent<HTMLSpanElement>,
+    ) => void;
+    className?: string;
+}
+
+export const Checkbox = forwardRef<HTMLInputElement, CheckboxProps>(
+    (
+        {
+            checked = false,
+            disabled = false,
+            indeterminate = false,
+            showInfoIcon = false,
+            infoIconClassName,
+            size = "sm",
+            label,
+            labelClassName,
+            name,
+            onChange,
+            className,
+            ...rest
+        },
+        ref,
+    ) => {
+        const [isChecked, setIsChecked] = useState(checked);
+        const [isIndeterminate, setIsIndeterminate] = useState(indeterminate);
+
+        useEffect(() => {
+            setIsChecked(checked);
+        }, [checked]);
+
+        useEffect(() => {
+            setIsIndeterminate(indeterminate);
+        }, [indeterminate]);
+
+        const onChangeHandler = (
+            e:
+                | React.ChangeEvent<HTMLInputElement>
+                | React.KeyboardEvent<HTMLSpanElement>,
+        ) => {
+            if (isIndeterminate) {
+                setIsIndeterminate(!isIndeterminate);
+                setIsChecked(false);
+            } else {
+                setIsChecked(!isChecked);
```

Comment:
```suggestion
                setIsChecked(prev => !prev);
```

Reply:
Thank you, but actually i'm setting new value inside of onChangeHandler, which will be called only onClick/onKeyDown, so I will work with the latest ones.

Reply:
It's better to update state in functional way if new state depends on the previous state.